src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1589,24 +1591,36 @@ else if (toUpdate.getContentAccessMode() != null) {
             changesMade = true;
         }
 
+        changesMade = changesMade || complianceChangesMade || sysPurposeChangesMade;
+
         if (changesMade) {
             log.debug("Consumer {} updated.", toUpdate.getUuid());
 
             // Set the updated date here b/c @PreUpdate will not get fired
             // since only the facts table will receive the update.
             toUpdate.setUpdated(new Date());
 
+            Event event = eventBuilder.setEventData(toUpdate).buildEvent();
+            sink.queueEvent(event);
+        }
+
+        if (sysPurposeChangesMade) {
             // this should update compliance on toUpdate, but not call the curator
-            complianceRules.getStatus(toUpdate, null, false, false);
             systemPurposeComplianceRules.getStatus(toUpdate, toUpdate.getEntitlements(), null, false);
+        }
 
-            Event event = eventBuilder.setEventData(toUpdate).buildEvent();
-            sink.queueEvent(event);
+        if (complianceChangesMade) {
+            // this should update compliance on toUpdate, but not call the curator
+            complianceRules.getStatus(toUpdate, null, false, false);
         }
 
         return changesMade;
     }
 
+    private boolean complianceFactsChanged(ConsumerDTO updated, Consumer toUpdate) {
+        return !Objects.equals(ComplianceFacts.filter(updated.getFacts()), ComplianceFacts.of(toUpdate));
```

Comment:
It's a bit unclear reading this that the `filter` and `of` methods are basically doing the same thing.
We could make the interface of that class a bit better by adding an overloaded `of(ConsumerDTO)` method, and make the `filter` method private.